spacy/cli/benchmark_speed.py

```diff
@@ -171,5 +173,7 @@ def print_outliers(sample: numpy.ndarray):
 def warmup(
     nlp: Language, docs: List[Doc], warmup_epochs: int, batch_size: Optional[int]
 ) -> numpy.ndarray:
-    docs = warmup_epochs * docs
-    return annotate(nlp, docs, batch_size)
+    warmup_docs = []
+    for _ in range(warmup_epochs):
+        warmup_docs += [doc.from_docs([doc]) for doc in docs]
+    return annotate(nlp, warmup_docs, batch_size)
```

Comment:
`Doc` has a `copy` method, so something like the following should work:

```suggestion
    return annotate(nlp, [doc.copy() for doc in docs], batch_size)
```